client/app/scripts/superdesk-archive/controllers/list.js

```diff
@@ -44,31 +45,49 @@ define([
         $scope.openUpload = function openUpload() {
             superdesk.intent('upload', 'media');
         };
-
         this.fetchItems = function fetchItems(criteria) {
             if (resource == null) {
                 return;
             }
             $scope.loading = true;
-            resource.query(criteria).then(function(items) {
+            resource.query(criteria)
+            .then(function(items) {
                 $scope.loading = false;
                 $scope.items = items;
+                $scope.page = 1;
             }, function() {
                 $scope.loading = false;
             });
         };
-
+        $scope.fetchNext = function() {
+            if (!$scope.fetching) {
+                $scope.fetching = true;
+                var resource = self.getResource();
+                $scope.page = $scope.page + 1;
+                var query = self.getQuery(_.omit($location.search(), '_id'), true);
+                query.from = ($scope.page - 1) * query.size;
+                var criteria = {source: query};
+                $scope.loading = true;
+                resource.query(criteria)
+                .then(function(items) {
+                    $scope.loading = false;
+                    $scope.items._items = $scope.items._items.concat(items._items);
+                    $scope.fetching = false;
+                }, function() {
+                    $scope.loading = false;
```

Comment:
`$scope.loading = false` repeats both in success and error handler - better to have that in `.finally()` to avoid repetition.